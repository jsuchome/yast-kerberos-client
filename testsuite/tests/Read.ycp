/**
 *  Read.ycp
 *  Test of Kerberos::Read function (whole read process, many config files)
 *  Author:	Jiri Suchomel <jsuchome@suse.cz>
 *  $Id$
 */

{

    // testedfiles: Kerberos.ycp Pam.ycp
    import "Testsuite";
    import "Kerberos";

    map READ = $[
	"etc": $[
	    // /etc/krb5.conf
	    "krb5_conf" : $[
		"v" : $[
		    "libdefaults": $[
			"default_realm": ["SUSE.CZ"],
			"clockskew": ["500"]
		    ],
		    "SUSE.CZ": $[
			"kdc": ["chimera.suse.cz", "kdc.suse.cz"],
			"admin_server": nil,
			"default_domain": nil,
		    ],
		    "pam" : $[
			"minimum_uid": ["1"],
			"renew_lifetime": nil,
			"ticket_lifetime": nil,
			"forwardable": nil,
			"proxiable": nil,
			"retain_after_close": nil,
			"use_shmem"	: [""],
			"use_authtok"		: nil,
		    ],
		    "pkinit" : $[
			"trusted_servers": nil,
		    ]
		],
	    ],
	    // /etc/security/pam_*
	    "security": $[
		"section": $[
		    "/etc/security/pam_unix2.conf": $[],
		    "/etc/security/pam_pwcheck.conf": $[]
		],
		"v": $[
		    "/etc/security/pam_unix2.conf": $[
			"passwd" : "nullok use_ldap",
			"auth"	: "nullok use_krb5"
		    ],
		]
	    ],
	    // /etc/ssh/ssh_config
	    "ssh" : $[
		"ssh_config": $[
		    "s": $[
			"*": $[]
		    ],
		    "v": $[
			"*": $[
			    "GSSAPIAuthentication" : "yes",
			    "GSSAPIDelegateCredentials": "yes"
			]
		    ],
		]
	    ],
	],
	"sysconfig": $[
	    "openafs-client": nil
	],
	"target": $[
	    "stat": $[ 1:2 ]
	]
    ];

    map EX = $[
	"target": $[
	    "bash": 0,
	    "bash_output": $[
// call of ypdomainname
//		"stdout" : "suse.cz"
// call of pam-config
		"stdout" : "password: "
	    ]
	]
    ];

    Testsuite::Dump ("==== reading... ============================");

    Testsuite::Test(``(Kerberos::Read ()), [ READ, $[], EX ], 0);

    Testsuite::Dump ("============================================");

    Testsuite::Dump ( sformat ("kerberos used: %1", Kerberos::use_pam_krb) );

    Testsuite::Dump ( sformat ("default realm: %1", Kerberos::default_realm) );
    Testsuite::Dump ( sformat ("kdc: %1", Kerberos::kdc) );

    Testsuite::Dump ( sformat ("\nssh support: %1", Kerberos::ssh_support) );
    Testsuite::Dump ("============================================");

//    READ ["etc", "krb5_conf", "v", "libdefaults", "default_realm"]	= nil;
    READ ["target", "stat"]						= $[];

    Testsuite::Dump ("==== reading... ============================");

    Testsuite::Test(``(Kerberos::Read ()), [ READ, $[], EX ], 0);

    Testsuite::Dump (sformat ("default realm: %1", Kerberos::default_realm) );

    Testsuite::Dump ("============================================");


}
