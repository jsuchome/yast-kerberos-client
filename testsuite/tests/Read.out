Dump	==== reading... ============================
Dir	.etc.security.section: ["/etc/security/pam_pwcheck.conf", "/etc/security/pam_unix2.conf"]
Read	.etc.security.v."/etc/security/pam_unix2.conf"."auth" "nullok use_krb5"
Dir	.etc.krb5_conf.s: []
Read	.etc.krb5_conf.v.libdefaults.default_realm "SUSE.CZ"
Read	.etc.krb5_conf.v.libdefaults.clockskew "500"
Read	.etc.krb5_conf.v."SUSE.CZ"."kdc" "chimera.suse.cz"
Read	.etc.krb5_conf.v."SUSE.CZ"."admin_server" nil
Read	.etc.krb5_conf.v."SUSE.CZ"."default_domain" nil
Read	.etc.krb5_conf.v.pam.ticket_lifetime nil
Read	.etc.krb5_conf.v.pam.renew_lifetime nil
Read	.etc.krb5_conf.v.pam.forwardable nil
Read	.etc.krb5_conf.v.pam.proxiable nil
Read	.etc.krb5_conf.v.pam.retain_after_close nil
Read	.etc.krb5_conf.v.pam.minimum_uid "1"
Read	.etc.krb5_conf.v.pam.try_first_pass "false"
Execute	.target.bash_output "/bin/ypdomainname" $["stdout":"suse.cz"]
Read	.sysconfig.afs-client.THIS_CELL nil
Read	.sysconfig.afs-client.THIS_CELL_SERVER_NAME nil
Read	.sysconfig.afs-client.THIS_CELL_SERVER nil
Read	.sysconfig.afs-client.REGENERATE_CELL_INFO nil
Dir	.etc.ssh.ssh_config.s: ["*"]
Execute	.target.bash_output "LANG=C /bin/hostname" $["stdout":"suse.cz"]
Dir	.etc.ssh.ssh_config.s: ["*"]
Dir	.etc.ssh.ssh_config.v."*": ["GSSAPIAuthentication", "GSSAPIDelegateCredentials"]
Read	.etc.ssh.ssh_config.v."*"."GSSAPIAuthentication" "yes"
Read	.etc.ssh.ssh_config.v."*"."GSSAPIDelegateCredentials" "yes"
Return	true
Dump	============================================
Dump	kerberos used: true
Dump	AFS used: false
Dump	
Dump	default domain: suse.cz
Dump	default realm: SUSE.CZ
Dump	kdc: chimera.suse.cz
Dump	
Dump	ssh support: true
Dump	============================================
Dump	==== reading... ============================
Read	.etc.security.v."/etc/security/pam_unix2.conf"."auth" "nullok use_krb5"
Execute	.target.bash "/usr/bin/touch /etc/krb5.conf" 0
Dir	.etc.ssh.ssh_config.s: ["*"]
Execute	.target.bash_output "LANG=C /bin/hostname" $["stdout":"suse.cz"]
Dir	.etc.ssh.ssh_config.s: ["*"]
Dir	.etc.ssh.ssh_config.v."*": ["GSSAPIAuthentication", "GSSAPIDelegateCredentials"]
Read	.etc.ssh.ssh_config.v."*"."GSSAPIAuthentication" "yes"
Read	.etc.ssh.ssh_config.v."*"."GSSAPIDelegateCredentials" "yes"
Return	true
Dump	default realm: SUSE.CZ
Dump	============================================
